controllers/front/ProductController.php

```diff
@@ -576,6 +580,60 @@ protected function getMinimalProductOrDeclinationQuantity($product)
         return $minimalProductQuantity;
     }
 
+    /**
+     * Displays information, if the customer has this product in cart already.
+     */
+    protected function addCartQuantityNotification()
+    {
+        if ((bool) Configuration::get('PS_DISPLAY_AMOUNT_IN_CART') !== true) {
+            return;
+        }
+
+        // Get quantity of this product in cart, it will return an array with
+        // quantity of this single product and also quantity in packs
+        $cartProductQuantity = $this->context->cart->getProductQuantityAllVariants(
+            $this->id_product
+        );
+
+        // If there is nothing in the cart for this product, nothing to do here
+        if ($cartProductQuantity['deep_quantity'] == 0) {
+            return;
+        }
+
+        // Now, we got from this function both total count and standalone count, works like this:
+        // If you have 1 standalone - quantity 1, deep_quantity 1
+        // If you have 1 standalone and 1 in pack - quantity 1, deep_quantity 2
+        // If you have 1 in pack - quantity 0, deep_quantity 1
+        $standaloneQuantity = (int) $cartProductQuantity['quantity'];
+        $quantityInPacks = 0;
+
+        // If we have some products in pack only, we will subtract it to
+        if ($cartProductQuantity['quantity'] != $cartProductQuantity['deep_quantity']) {
+            $quantityInPacks = (int) $cartProductQuantity['deep_quantity'] - (int) $cartProductQuantity['quantity'];
+        }
+
+        // Render nice notifications so the user knows what is happening
+        if ($standaloneQuantity > 0 && $quantityInPacks > 0) {
+            $this->info[] = $this->trans(
+                'You have %1s piece(s) of this product in cart and another %2s piece(s) included in packs.',
```

Comment:
Is it possible to add something like but replace the (product name) comment with a variable that fetches the name of the current product displayed on page?

Reply:
I don't think it's required to show the product name if the message is related to the product the customer is viewing right now. 👍Plus your wording is weird imho.

Reply:
Product name can be very long. I think also it adds some complexity in the warning

Reply:
Another problem is other languages. We have different word endings for each grammatical case + plurals.

English: Hummingbird printed sweater
English: Your cart contains 5 Hummingbird printed sweater, and 1 Hummingbird printed sweater are included in packs.

Czech: Váš košík obsahuje 5 Hummingbird printed sweaterů a 1 Hummingbird printed sweater máte v balíčcích.

It looks bad even in English when you don't have the plural.

Reply:
Hi, 
Thanks @l-delin @Hlavtox for your propositions. 
As we're trying to be specific in a context where we just add complementary informations to avoid friction for users and avoid them to check their cart directly it's only faire that we're also being specific about what we talk about.

If we announce a quantity and we know a quantity of what, then there's a sense in adding it in the formulation rather than being vague with "product", even though we could definitely argue that users know what product we're talking about.

As content relies on grammar, syntax, semantics and context, let's try not to evaluate content based on feeling, such as it being weird or not, but rather evaluate its efficiency regarding a specific goal. 

On a language basis, "a piece of product" doesn't necessarily design a unit but rather a part of a product (as if you'd separate it).
 So we'll go with something that is at least correct in english. 
 
 Thanks !

Reply:
We all agree that the "piece" is not ideal wording, but I don't think using the product name will be suitable in this context.

Reply:
An update:
I will push the review anyway to have the feature, hopefully for 8.1.2. In parallel, I'm planning a quick discussion with @axelvrsc so we can find a suitable candidate. If in the meantime, the feature is merged. We can still improve it later.

Reply:
pay/buy/sell by the piece - COMMERCE to pay for, buy, or sell something in single units rather than in groups

Reply:
> If the users is viewing the product page and the message says YOUR CART CONTAINS 2 PIECES OF THIS PRODUCT, the user must be an complete idiot if he doesn't know what we are talking about. 👍

"Piece" can be used on too many different contexts, even translator could get it wrong.

Reply:
@MatShir Oh OK, I wrote it thinking somebody is putting his nose into things! :D

**To summarize:**
- Product name would be nice sometimes, but we are unable to do it properly, even in English.  It can be long and ugly. It would require joining some more tables.
- We have a czech word "kus"/"kusy" which means how many articles are there to purchase in this context. The direct translation is "pieces" and I didn't find any other english word that would mean the same thing. I posted some things above. :-)

Reply:
My original wording is the best I think 🤷

![Snímek obrazovky 2023-09-14 151811](https://github.com/PrestaShop/PrestaShop/assets/6097524/1be08677-e1fa-45fc-b8ae-50afb7684379)

Reply:
Option 3 appears more aesthetically pleasing. It addresses the concern regarding the "piece" and is in proximity to the success message, aligning with the overall context.

Reply:
Hello, yes. I think ultimately we can use the following as a success message on the FO product page:

"You've successfully added %1s products to your cart."
"You've successfully added %1s products included in packs in your cart."
"You've successfully added %1s products and %2s products included in packs in your cart."

Any improvement can be made later. Hopefully your ICU can fix the product/products issue so we don't have to worry about it anymore @Hlavtox

Reply:
You're right @MatShir sorry @Hlavtox I forgot to introduce myself, my mistake! 
I'm Axel, Lea and Julie's manager and will be working with Julie on the wording contributions now that Lea will not be working with us from tomorrow. 

Thanks for your exchanges and the details you put. 
I'm glad we found a first wording to start with. As @l-delin said we'll see later if improvement is needed.

Thanks again @Hlavtox for your contribution, looking forward to work on other wordings with you!

Reply:
@l-delin It's not a success message, it's displayed after you revisit the product page. `You've successfully added bla bla` doesn't really fit.

This is how it works:
https://github.com/PrestaShop/PrestaShop/assets/6097524/f197cf6c-7d91-4db8-b5dd-5dac3bd36b79

Reply:
If you want to avoid the pieces, I have thought of this now:
- Your cart contains 1 of these products.
- Your cart contains 5 of these products.
- Your cart contains 5 of these products and another 5 are part of packs in your cart.
-5 of these products are part of packs in your cart.

WDYT???

Reply:
Hey @Hlavtox I like the last proposition. We can go with this.